### PR TITLE
Support Network Commissioning for Infineon P6 platform

### PR DESCRIPTION
**Pull request by praveenCY**
#### Problem
* Network Commissioning not supported for Infineon P6
* Persistent Storage app not supported for Infineon P6

#### Change overview
* Add NetworkCommissioningDriver.h and NetworkCommissioningWiFiDriver.cpp for P6 
    - New NetworkCommissioning flow is used for all Infineon P6 example apps
* Add Persistent Storage app for P6

#### Testing
* Tested with chiptool
* Verified Commissioning, Scan-networks, AddorUpdateWiFi, attributes of Network Commissioning Clusters like max-networks, networks etc

### CODE REVIEW

File: examples/persistent-storage/p6/.gn [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2021 Project CHIP Authors
```
Comment (andy31415) [inline examples/persistent-storage/p6/.gn:1]: Persistent-storage  example app does not seem to match PR description (support network commissionig).

Could we split the PRs?

Comment (woody-apple) [reply examples/persistent-storage/p6/.gn:1]: Feels like we can follow up on this later?

**Comment by woody-apple** [comment]
Fast tracking platform changes by a platform maintainer.

**Pull request merged**